### ISSUE TITLE
Simple repeater in relationship presents as [object Object]

### ISSUE THREAD

**Issue by mdpoulter** [laravel-filament/filament#10313]
### Package

filament/filament

### Package Version

v3.1.21

### Laravel Version

v10.37.3

### Livewire Version

v3.3.0

### PHP Version

PHP 8.2.13

### Problem description

Using a simple repeater in a relationship layout component shows `[object Object]` instead of the text for each item in the array. This does not happen if the field is located on the model itself (left), only if it's nested in a relationship (right).

<img width="1250" alt="Screenshot 2023-12-14 at 22 36 54" src="https://github.com/filamentphp/filament/assets/1091085/306c85ad-9378-4227-a89a-e4e6343a491c">

The issue is similar, but does not appear to be the same as #8874

### Expected behavior

The text from the array element should be displayed.

### Steps to reproduce

Create a simple repeater within a nested relationship layout component:

```php
// Works correctly
Forms\Components\Repeater::make('phone_numbers')
    ->simple(
        TextInput::make('phone_number')
            ->required(),
    ),

// Presents filled lines as [object Object]
Forms\Components\Group::make()
    ->relationship('address')
    ->schema([
        Forms\Components\Repeater::make('lines')
            ->simple(
                TextInput::make('line')
                    ->required(),
            ),
    ]),
```

### Reproduction repository

https://github.com/mdpoulter/filament-issue-repeater

**Comment by mdpoulter** [comment]
@danharrin Just a heads up that this seems to be broken again in the latest release. It looks like it may have been caused by #10537 (which seems to have fixed parts of the other issue I created #10536 😄 ).

I haven't created a new reproduction repo, but can if you would like.

**Comment by danharrin** [comment]
Uh oh, yes please if you could create a reproduction repo

**Comment by danharrin** [comment]
unless the existing one works for the new issue as well? just confirm if that is the case

**Comment by mdpoulter** [comment]
I've created the reproduction for this here using the latest version: https://github.com/mdpoulter/filament-issue/tree/repeater

<img width="1251" alt="Screenshot 2023-12-29 at 16 36 56" src="https://github.com/filamentphp/filament/assets/1091085/bf38a46d-44ec-49d1-98ab-ff0b4a77d3a0">

**Comment by danharrin** [comment]
Thanks for your patience. #10537 seems to have simplified the situation and I was able to fix the problem much easier in #10764
